try to work without std::make_unique [skip vbump] (#354)
openpharma/mmrm: vignettes/package_structure.Rmd

### `Makevars`

This file specifies additional flags used in the source code compilation. 

> Also we ask `TMB` to use the `TMB` automatic differentiation framework (instead of the default `CppAD` framework) to be more efficient, and disable useless warnings from the `Eigen` library compilation.

Reviewer: ```suggestion
W ask `TMB` to use the `TMB` automatic differentiation framework (instead of the default `CppAD` framework) to be more efficient, and disable useless warnings from the `Eigen` library compilation.
```
